code/modules/ruins/lavaland_ruin_code.dm

```diff
@@ -0,0 +1,213 @@
+	user.visible_message("<span class='danger'><B>The ground splits beneath [user] as their hand leaves the mirror!</B></span>")
+	var/turf/T = get_turf(user)
+	T.ChangeTurf(/turf/simulated/chasm/straight_down)
+	var/turf/simulated/chasm/straight_down/C = T
+	C.drop(user)*/
+
+//Sloth - I'll finish this item later
+
+//Envy
+
+/obj/item/weapon/knife/envy
+	name = "envy's knife"
+	desc = "Their success will be yours."
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "render"
+	item_state = "render"
+	force = 18
+	throwforce = 10
+	w_class = 3
+	hitsound = 'sound/weapons/bladeslice.ogg'
+/*TODO: Attack
+/obj/item/weapon/knife/envy/afterattack(atom/movable/AM, mob/living/carbon/human/user, proximity)
+	..()
+	if(!proximity)
+		return
+	if(!istype(user))
+		return
+	if(istype(AM, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = AM
+		if(user.real_name != H.dna.real_name)
+			user.real_name = H.dna.real_name
+			H.dna.transfer_identity(user, transfer_SE=1)
+			user.updateappearance(mutcolor_update=1)
+			user.domutcheck()
+			user << "You assume the face of [H]. Are you satisfied?"*/
+
+///Ash Walkers
+
+/mob/living/simple_animal/hostile/spawner/ash_walker
+	name = "ash walker nest"
+	desc = "A nest built around a necropolis tendril. The eggs seem to grow unnaturally fast..."
+	icon = 'icons/mob/nest.dmi'
+	icon_state = "ash_walker_nest"
+	icon_living = "ash_walker_nest"
+	health = 200
+	maxHealth = 200
+	loot = list(/obj/effect/gibspawner, /obj/item/device/assembly/signaler/anomaly)
+	del_on_death = 1
+	var/meat_counter
+
+/mob/living/simple_animal/hostile/spawner/ash_walker/Life()
+	..()
+	if(!stat)
+		consume()
+		spawn_mob()
+
+/mob/living/simple_animal/hostile/spawner/ash_walker/proc/consume()
+	for(var/mob/living/H in view(src,1)) //Only for corpse right next to/on same tile
+		if(H.stat)
+			visible_message("<span class='warning'>Tendrils reach out from \the [src.name] pulling [H] in! Blood seeps over the eggs as [H] is devoured.</span>")
+			playsound(get_turf(src),'sound/magic/Demon_consume.ogg', 100, 1)
+			meat_counter ++
+			H.gib()
+
+/mob/living/simple_animal/hostile/spawner/ash_walker/spawn_mob()
+	if(meat_counter >= 2)
+		new /obj/effect/landmark/corpse/mob_spawn/ash_walker(get_step(src.loc, SOUTH))
+		visible_message("<span class='danger'>An egg is ready to hatch!</span>")
+		meat_counter -= 2
+
+/obj/effect/landmark/corpse/mob_spawn/ash_walker
+	name = "ash walker egg"
+	icon = 'icons/mob/lavaland/lavaland_monsters.dmi'
+	icon_state = "large_egg"
+	mob_species = /datum/species/unathi/ashwalker
+	corpsehelmet = /obj/item/clothing/head/helmet/gladiator
+	corpseuniform = /obj/item/clothing/under/gladiator
+	anchored = 0
+	density = 0
+	flavour_text = {"<B>You are an Ash Walker. Your tribe worships<span class='danger'>the necropolis</span>. The wastes are sacred ground, it's monsters a blessed bounty. You have seen lights in the distance though, the arrival of outsiders seeking to destroy the land. Fresh sacrifices.</B>"}
```

Comment:
"You are an Ash Walker. Your tribe worships <span class='danger'>the necropolis</span>. The wastes are sacred ground, its monsters a blessed bounty. You have seen lights in the distance though; the arrival of outsiders seeking to destroy the land. Fresh sacrifices."
 
Slight grammar changes for readability.